### PR TITLE
env var only config

### PR DESCRIPTION
**Pull request by eschultink**
### Features
  - only allow some config as env vars

### Change implications

 - dependencies added/changed? **no**
 - something important to note in future release notes? **Yes**
   - NOTE in `CHANGELOG.md` anything that will show up in `terraform plan`/`apply` that isn't
     obviously a no-op? **if anyone is depending on overriding env vars with secrets/parameter store, this will break that**
   - breaking changes? if in module/example that is NOT marked `alpha`, requires major version
     change

### CODE REVIEW

File: java/core/src/test/java/co/worklytics/psoxy/gateway/ProxyConfigPropertyTest.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        strings = {
+            "PSOXY_ENCRYPTION_KEY",
+            "ENCRYPTION_KEY_IP",
+            "RULES",
+            "PSOXY_SALT",
+            "SALT_IP",
+        }
+    )
+    @ParameterizedTest
+    public void remoteConfigVars(String paramName) {
```
Comment (jlorper) [inline java/core/src/test/java/co/worklytics/psoxy/gateway/ProxyConfigPropertyTest.java:23]: :red_circle: missing test body

File: java/impl/gcp/src/main/java/co/worklytics/psoxy/SecretManagerConfigService.java [hunk 2]
```diff
@@ -61,6 +61,10 @@ public SecretManagerConfigService(@Assisted("projectId") @NonNull String project
 
     @Override
     public void putConfigProperty(ConfigProperty property, String value) {
+        if (property.isEnvVarOnly()) {
```
Comment (jlorper) [inline java/impl/gcp/src/main/java/co/worklytics/psoxy/SecretManagerConfigService.java:64]: Tip: I like preconditions for these kind of checks.
`Preconditions.checkArgument(!property.isEnvVarOnly(), "Can't put env-only config property: " + property);`

**Review by jlorper — CHANGES_REQUESTED** [review]
missing test body

**Review by aperez-worklytics — APPROVED** [review]
OK after @jlorper changes

**Review by jlorper — APPROVED** [review]

**Pull request merged**